internals/src/iosMain/kotlin/at/asitplus/signum/internals/InteropUtils.kt

```diff
@@ -3,31 +3,18 @@
 package at.asitplus.signum.internals
 
 import kotlinx.cinterop.*
-import platform.CoreFoundation.CFDictionaryCreateMutable
-import platform.CoreFoundation.CFDictionaryGetValue
-import platform.CoreFoundation.CFDictionaryRef
-import platform.CoreFoundation.CFDictionarySetValue
-import platform.CoreFoundation.CFErrorRefVar
-import platform.CoreFoundation.CFMutableDictionaryRef
-import platform.CoreFoundation.CFRelease
-import platform.CoreFoundation.CFTypeRef
-import platform.CoreFoundation.kCFBooleanFalse
-import platform.CoreFoundation.kCFBooleanTrue
-import platform.CoreFoundation.kCFTypeDictionaryKeyCallBacks
-import platform.CoreFoundation.kCFTypeDictionaryValueCallBacks
-import platform.Foundation.CFBridgingRelease
-import platform.Foundation.CFBridgingRetain
-import platform.Foundation.NSData
-import platform.Foundation.NSError
-import platform.Foundation.create
+import platform.CoreFoundation.*
+import platform.Foundation.*
 import platform.posix.memcpy
 import kotlin.experimental.ExperimentalNativeApi
 import kotlin.native.ref.createCleaner
 
 fun NSData.toByteArray(): ByteArray = ByteArray(length.toInt()).apply {
-    usePinned {
-        memcpy(it.addressOf(0), bytes, length)
-    }
+    if(length>Int.MAX_VALUE.toULong()) throw IndexOutOfBoundsException("length is too large")
```

Comment:
this check happens after you have already passed it to the `ByteArray` constructor, is this intentional?

Reply:
fixed